### PR TITLE
docs: initial documentation

### PR DESCRIPTION
**Pull request by israelidanny**
**Current Status:**
This is still a draft. I'm planning to finish an initial pass documenting most of the actions by EOD Jan 22nd and then move on to adding inline code examples and more detailed usage documentation as a second pass after which I'll switch the PR to a ready for review mode.

Pull request to track typedoc documentation.

Some changes not related to docs for review:
- I added a `doc:watch` command to be able to do `yarn doc:watch` and write docs easily
- I am removing `I` prefixes from interfaces as I encounter them (`ICancelBidParams` becomes `CancelBidParams`) because we had a mixed bag and TypeScript mostly discourages prefixed interfaces
- I am adding export to all the nested object parameter interfaces because typedoc doesn't include non-exported interfaces and types by default in the documentation. This is also good because end-users will be able to use those interfaces in their integrations of the JS SDK

### CODE REVIEW

File: src/actions/initStore.ts [hunk 1]
```diff
@@ -4,22 +4,33 @@ import { Connection } from '../Connection';
...
 
-interface IInitStoreParams {
+/** Parameters for {@link initStore} **/
+export interface InitStoreParams {
   connection: Connection;
+  /** Administrator wallet for the store **/
   wallet: Wallet;
+  /**
+   * - `true`: anyone can list on the store
+   * - `false`: only whitelisted creators can list on the store
```
Comment (aheckmann) [inline src/actions/initStore.ts:14]: please add a link to https://docs.metaplex.com/architecture/deep_dive/metaplex#whitelistedcreator

File: src/actions/initStore.ts [hunk 2]
```diff
@@ -4,22 +4,33 @@ import { Connection } from '../Connection';
...
-interface IInitStoreResponse {
+export interface InitStoreResponse {
   storeId: PublicKey;
   txId: string;
 }
 
+/**
+ * Initialize a {@link Store} account.
+ * This action will get a {@link Store} program derived account address for the provided wallet and initialize a store with that address, setting the given `wallet` as the admin
+ * @deprecated This action is deprecated, please use {@link initStoreV2} instead
```
Comment (aheckmann) [inline src/actions/initStore.ts:27]: Is this deprecation correct? I don't see deprecation warnings in MPL or docs yet.

File: src/actions/placeBid.ts [hunk 3]
```diff
@@ -16,29 +16,38 @@ import { getCancelBidTransactions } from './cancelBid';
...
 
-interface IPlaceBidResponse {
+export interface PlaceBidResponse {
   txId: TransactionSignature;
   bidderPotToken: PublicKey;
   bidderMeta: PublicKey;
 }
 
+/**
+ * Place a bit by taking it from the provided wallet and placing it in the bidder pot account.
```
Comment (aheckmann) [inline src/actions/placeBid.ts:42]:
```suggestion
 * Place a bid by taking it from the provided wallet and placing it in the bidder pot account.
```

File: src/actions/initStore.ts (hunk 2 above)
Comment (israelidanny) [reply src/actions/initStore.ts:27]: My bad. I kind of made an assumption here, but realized later on that V2 is still WIP. I'll remove the deprecation notice until V2 is production ready. Should I somehow state V2 to be not ready yet?

**Review by aheckmann — APPROVED** [review]

**Comment by israelidanny** [comment]
Yes , this was in the plan but then this was frozen and I didn’t get
updates. It should 100% be updated everywhere.

Token is an absolutely incorrect name for it. It’s an account. Naming it
token can make it seem like you’re expecting a token mint, when in reality
you’re to provide the account where this token will be stored.

**Pull request merged**